### DISCUSSION TITLE
error with "make test"

### DISCUSSION THREAD

**Discussion by ooggss** [apache/iceberg-rust#467] [Q&A] [answered]
I'm new to this project, and when I used docker to start all the tests of this project(without any change) , I got error below:
error: 5 targets failed:
    `-p iceberg --test file_io_s3_test`
    `-p iceberg-catalog-glue --test glue_catalog_test`
    `-p iceberg-catalog-hms --test hms_catalog_test`
    `-p iceberg-catalog-rest --test rest_catalog_test`
    `-p iceberg-datafusion --test integration_datafusion_hms_test`
make: *** [Makefile:52: test] Error 101

Is this my fault or it's just the way it is?

Here is my Dockerfile:
FROM rust:1.77.1

RUN apt-get update && apt-get install -y make

WORKDIR /usr/src/myapp

COPY . .

CMD ["./target/release/myapp"]

Here is the project version:
root@83b48b7f22d1:/usr/src/myapp# git log
commit 070576bfb47a6a7205026c9dbfb3397a06cc2986 (HEAD -> main, origin/main, origin/HEAD)
Author: Shabana Baig <43451943+s-akhtar-baig@users.noreply.github.com>
Date:   Thu Jun 13 15:01:34 2024 -0400

**Comment by odysa** [comment]
I think we need Docker to run these tests.
https://github.com/apache/iceberg-rust/blob/main/crates/catalog/rest/testdata/rest_catalog/docker-compose.yaml

Were these docker composes running when tests failed?

**Comment by liurenjie1024** [comment]
Yes, I think @odysa is right about it, integration test requires docker compose. You can run `make unit-test` to skip integration tests.

**Comment by ooggss** [comment]
Thanks for all your reply!

Does it mean that if I only want to determined whether the target function is correct or not, I just need to run `make unit-test` ?

**Comment by liurenjie1024** [comment]
> Thanks for all your reply!
> 
> Does it mean that if I only want to determined whether the target function is correct or not, I just need to run `make unit-test` ?

What do you mean by target function? `make unit-test` runs tests with integration tests excluded.

**Comment by ooggss** [comment]
> > Thanks for all your reply!
> > Does it mean that if I only want to determined whether the target function is correct or not, I just need to run `make unit-test` ?
> 
> What do you mean by target function? `make unit-test` runs tests with integration tests excluded.

I want to evaluate my tool, which is used to automatically translate code written in other language to rust(in function level). And I found that iceberg has been implemented in different language including rust. So I got some function pairs from iceberg, one of which is written in language A and the other is writted in rust, but they do the same thing. And I use my tool to translate the function from language A to rust, but I I need to determine whether the functions implemented by the translated function is consistent with the function in the rust version. So I need to replace the function in rust version and execute unit test to determined whether the target function is correct or not.

**Comment by liurenjie1024** [comment, marked as answer]
> > > Thanks for all your reply!
> > > Does it mean that if I only want to determined whether the target function is correct or not, I just need to run `make unit-test` ?
> > 
> > 
> > What do you mean by target function? `make unit-test` runs tests with integration tests excluded.
> 
> I want to evaluate my tool, which is used to automatically translate code written in other language to rust(in function level). And I found that iceberg has been implemented in different language including rust. So I got some function pairs from iceberg, one of which is written in language A and the other is writted in rust, but they do the same thing. And I use my tool to translate the function from language A to rust, but I I need to determine whether the functions implemented by the translated function is consistent with the function in the rust version. So I need to replace the function in rust version and execute unit test to determined whether the target function is correct or not.

This depends on what your target function is generated. Usually integration tests lives in test folder of a crate, such as [this one](https://github.com/apache/iceberg-rust/blob/9b8f7913e8b3136f346ab9aea60ab5c2512c5f0d/crates/catalog/rest/tests), and unit-tests live in src directory of crates, such as [this one](https://github.com/apache/iceberg-rust/blob/0937f19d4c4b4ea14d423e6c78731904b4c9012e/crates/catalog/rest/src/catalog.rs#L666). You can learn more about [test organizaiton in rust](https://doc.rust-lang.org/book/ch11-03-test-organization.html) .

**Comment by ooggss** [comment]
> > > > Thanks for all your reply!
> > > > Does it mean that if I only want to determined whether the target function is correct or not, I just need to run `make unit-test` ?
> > > 
> > > 
> > > What do you mean by target function? `make unit-test` runs tests with integration tests excluded.
> > 
> > 
> > I want to evaluate my tool, which is used to automatically translate code written in other language to rust(in function level). And I found that iceberg has been implemented in different language including rust. So I got some function pairs from iceberg, one of which is written in language A and the other is writted in rust, but they do the same thing. And I use my tool to translate the function from language A to rust, but I I need to determine whether the functions implemented by the translated function is consistent with the function in the rust version. So I need to replace the function in rust version and execute unit test to determined whether the target function is correct or not.
> 
> This depends on what your target function is generated. Usually integration tests lives in test folder of a crate, such as [this one](https://github.com/apache/iceberg-rust/blob/9b8f7913e8b3136f346ab9aea60ab5c2512c5f0d/crates/catalog/rest/tests), and unit-tests live in src directory of crates, such as [this one](https://github.com/apache/iceberg-rust/blob/0937f19d4c4b4ea14d423e6c78731904b4c9012e/crates/catalog/rest/src/catalog.rs#L666). You can learn more about [test organizaiton in rust](https://doc.rust-lang.org/book/ch11-03-test-organization.html) .

I got it! Thank you so much!

I think that I should execute all tests (including integration test)
